src/provider/mod.rs

```diff
@@ -601,6 +594,36 @@ impl RpcHandler {
     }
 }
 
+/// Create data sources from a path.
+pub fn create_data_sources(root: PathBuf) -> anyhow::Result<Vec<DataSource>> {
```

Comment:
does this have to be pub?  not sure i like this in the provider api, though we could always move it later

Reply:
I was going crazy trying to keep things private. I need several internals for the resume test. I think we should reduce surface area again once we got a better way to do testing.